ARE/components/processor.serialport/src/main/java/eu/asterics/component/processor/serialport/SerialPortInstance.java

```diff
@@ -217,14 +220,16 @@ public Object setRuntimePropertyValue(String propertyName, Object newValue) {
         }
         if ("cimId".equalsIgnoreCase(propertyName)) {
             final Object oldValue = propCimId;
-            String s = newValue.toString();
-            if(s.contains("0x")) {
-                s = s.substring(2);
-            }
-            try {
-                propCimId = (short) Integer.parseInt(s, 16);
-            } catch (NumberFormatException e) {
-                logger.warning("could not format cimId, value is: " + s);
+            if(newValue != null) {
+                String s = newValue.toString().trim();
```

Comment:
What I forgot: 
after trimming, the value should be converted to lower or uppercase, to also support mixed versions of lowercase or uppercase hex strings.